lib/proxy.js

```diff
@@ -380,43 +381,47 @@ ReverseProxy.prototype.resolve = function(host, url){
   var route;
   for(var i=0; i < this.resolvers.length; i++){
     route = this.resolvers[i].call(this, host, url);
-    if(route && (_.isString(route) || _.isObject(route))){
+    if(route && (route = ReverseProxy.buildRoute(route))){
       //ensure resolved route has path that prefixes URL
       //no need to check for native routes.
-      route = buildRoute(route);
-      if(!route.isResolved || startsWith(url, route.pathname)){
+
+      if(!route.isResolved || route.path === '/' || startsWith(url, route.path)){
         return route;
       }
     }
   }
 };
 
-var buildRoute = function (route) {
+ReverseProxy.buildRoute = function (route) {
+  if(!_.isString(route) && !_.isObject(route)){
+    return null;
+  }
 
-  if(_.isObject(route) && route.hasOwnProperty('urls') && route.hasOwnProperty('pathname')){
+  if(_.isObject(route) && route.hasOwnProperty('urls') && route.hasOwnProperty('path')){
     //default route type matched.
     return route;
   }
 
   var cacheKey = _.isString(route) ? route : hash(route);
   if(routeCache.hasOwnProperty(cacheKey)){
+    routeCache[cacheKey]._hits++;
     return routeCache[cacheKey];
```

Comment:
regarding the cache. Should'nt we need to put a limit of cached keys? isn't there a risk that the cache could become too big overtime?